packages/@sanity/form-builder/src/sanity/DefaultAssetSource/AssetRow.tsx

```diff
@@ -0,0 +1,403 @@
+import type {Subscription} from 'rxjs'
+import React, {useRef, useState} from 'react'
+import styled from 'styled-components'
+import {Button, Box, Card, Flex, Stack, Label, Text, Tooltip, Grid, useToast} from '@sanity/ui'
+import {DocumentIcon, ChevronUpIcon, ChevronDownIcon, LinkIcon, TrashIcon} from '@sanity/icons'
+import {useTimeAgo} from '@sanity/base/hooks'
+import prettyBytes from 'pretty-bytes'
+import {Asset as AssetType} from '@sanity/types'
+import {versionedClient} from '../versionedClient'
+import {AssetUsageDialog} from './AssetUsageDialog'
+import {AssetMenu} from './AssetMenu'
+import {AssetMenuAction} from './types'
+import {formatMimeType} from './utils/mimeType'
+
+interface RowProps {
+  isMobile?: boolean
+  asset?: AssetType
+  isSelected?: boolean
+  onClick?: (...args: any[]) => any
+  onKeyPress?: (...args: any[]) => any
+  onDeleteFinished?: (...args: any[]) => any
+}
+
+const CardIconWrapper = styled.span`
+  background-color: transparent;
+  flex-shrink: 0;
+`
+
+// These are here because using vanilla UI components caused a type issue inside of styled-components
+const CustomFlex = styled(Flex)``
+const CustomText = styled(Text)``
+const CustomCard = styled(Card)<RowProps>`
+  &:hover-within ${CustomText} {
+    --card-muted-fg-color: var(--card-muted-fg-color);
+    --card-fg-color: var(--card-fg-color);
+  }
+
+  ${(props) =>
+    props.isSelected &&
+    `
```

Comment:
would be good to use the [`css` template tag](https://styled-components.com/docs/api#css) here so that editors can provide syntax highlighting